### ISSUE TITLE
Slow speed on NVIDIA Quadro RTX 4000 using latest commit (3532d8de4f8c3559c7a6f123a1b21bab69a66d63)

### ISSUE THREAD

**Issue by vshcherbyna** [glinscott/nnue-pytorch#211]
Hello there,

Using nnue-pytorch to train networks on old architecture (halfkp_256x2-32-32) I used to see speeds of roughly 30 iterations per second on my NVIDIA Quadro RTX 4000. 

Using the same machine/GPU after upgrading to latest commit on nnue-pytorch (3532d8de4f8c3559c7a6f123a1b21bab69a66d63) I see speeds of 2.6 iterations per second:

![image](https://user-images.githubusercontent.com/337059/202919544-eb771029-1158-432a-9bc6-b0a5f9927318.png)

Can you please tell if this is expected? When I start the training I see a big spike of GPU usage for a few seconds (30-100%)  but then it calms down.

I am running Windows 11 x64 with NVIDIA Quadro RTX 4000 and latest CUDA toolkit.

I made sure I am not using CPU training by modifying the following line in train.py file:

`main_device = 'cuda:' + str(trainer.strategy.root_device.index)`

I run training in the following way:

`python train.py --smart-fen-skipping --random-fen-skipping 3  --batch-size 16384 --threads 2 --num-workers 2 --gpus 1 "data.binpack" "val.binpack"
`

I tried various parameters for threads or num-workers it does not change a thing.

Thank you for any hint/help on the issue!

Best regards, Volodymyr.

**Comment by Sopel97** [comment]
Some speed loss is expected but not that much I think. Could you bisect the commits to find where the biggest speed losses happen?

**Comment by vshcherbyna** [comment]
After further analysis on my side I determined that the issue was caused by using a --num-workers with a very small value. Once I increased it the speed went up and it is identical between the master and 3532d8de4f8c3559c7a6f123a1b21bab69a66d63 commit.

Sorry for the noise.
